src/braket/circuits/circuit.py

```diff
@@ -1305,18 +1309,41 @@ def _get_frames_waveforms_from_instrs(
                     _validate_uniqueness(waveforms, waveform)
                     waveforms[waveform.id] = waveform
             # this will change with full parametric calibration support
-            elif hasattr(type(instruction.operator), "angle"):
+            elif isinstance(instruction.operator, Parameterizable):
                 if gate_calibrations is not None:
-                    key = (type(instruction.operator)(FreeParameter("theta")), instruction.target)
-                    if key in gate_calibrations.calibration_data:
-                        ps = gate_calibrations.get_gate_calibration(key)
-                        bound_key = (
-                            type(instruction.operator)(instruction.operator.angle),
-                            instruction.target,
-                        )
-                        gate_calibrations._calibration_data[bound_key] = ps(
-                            theta=instruction.operator.angle
-                        )
+                    for key, calibration in gate_calibrations.calibration_data.copy().items():
```

Comment:
Can we do a lookup using` instruction.operator` and `instruction.target` for comparison?

Reply:
```
a = Instruction(Gate.Rx(1), [0, ])
b = Instruction(Gate.Rx(FreeParameter("theta")), [0, ])
print(a.operator == Gate.Rx(angle=1))
print(a.operator == Gate.Rx(angle=2))
print(b.operator == Gate.Rx(FreeParameter("theta")))


>>> True
>>> False
>>> True
```
Did some quick testing. 

```suggestion
                    if gate_calibration.get((instruction.operator, QubitSet(instruction.target)) is not None and 
                    all(not isinstance(param, float) for param in instruction.operator.parameters):
```
We already check if the operator has take parameters so here we check to see if there are any unbound params. For gates that take in multiple params, we check all the params to see if 1 or more are floats. 

One assumption I am making is that the unbound params for `instruction` are the same for unbound params for `keys`. If the goal is to find the unbound parameters for `key` and bind for `instruction` params, then I think we can simplify the logic. In this case, I can push a different suggestion.